examples/exchange-files-in-browser/public/js/app.js

```diff
@@ -119,93 +120,44 @@ function onDrop (event) {
     onError('IPFS must be started before files can be added')
     return
   }
-  const dt = event.dataTransfer
-  const files = dt.files
-
-  function readFileContents (file) {
-    return new Promise((resolve) => {
-      const reader = new window.FileReader()
-      reader.onload = (event) => resolve(event.target.result)
-      reader.readAsArrayBuffer(file)
-    })
-  }
 
-  let filesArray = []
-  for (let i = 0; i < files.length; i++) {
-    filesArray.push(files[i])
+  let files = []
+  for (let i = 0; i < event.dataTransfer.files.length; i++) {
+    files.push(event.dataTransfer.files[i])
   }
 
-  filesArray.map((file) => {
-    readFileContents(file)
-      .then((buffer) => {
-        let fileSize = buffer.byteLength
-
-        if (fileSize < 50000000) {
-          return node.files.add([{
-            path: file.name,
-            content: new node.types.Buffer(buffer)
-          }])
-        } else {
-          // use createAddStream and chunk the file.
-          let progress = 0
-
-          let myReadableStreamBuffer = new streamBuffers.ReadableStreamBuffer({
-            // frequency: 10,   // in milliseconds.
-            chunkSize: 32048  // in bytes.
-          })
-
-          node.files.createAddStream((err, stream) => {
-            if (err) throw err
-
-            stream.on('data', (file) => {
-              $multihashInput.value = file.hash
-              $filesStatus.innerHTML = `Added ${file.path} as ${file.hash}`
-
-              if (progressbar) {
-                clearInterval(progressbar)
-                progress = 0
-              }
-            })
-
-            myReadableStreamBuffer.on('data', (chunk) => {
-              progress += chunk.byteLength
-            })
-
-            if (!myReadableStreamBuffer.destroy) {
-              myReadableStreamBuffer.destroy = () => {}
-            }
-
-            stream.write({
-              path: file.name,
-              content: myReadableStreamBuffer
-            })
-
-            myReadableStreamBuffer.put(Buffer.from(buffer))
-            myReadableStreamBuffer.stop()
-
-            myReadableStreamBuffer.on('end', () => {
-              stream.end()
-            })
-
-            myReadableStreamBuffer.resume()
-
-            // progress.
-            let progressbar = setInterval(() => {
-              console.log('progress: ', progress, '/', fileSize, ' = ', Math.floor((progress / fileSize) * 100), '%')
-            }, 5000)
-          })
+  pull(
+    pull.values(files),
+    pull.through((file) => console.log('Adding %s', file)),
+    pull.asyncMap((file, cb) => pull(
+      pull.values([{
+        path: file.name,
+        content: pullFilereader(file)
+      }]),
+      node.files.createAddPullStream(),
```

Comment:
This is awesome, but we shouldn't have examples with undocumented (and already agreed to change) API

Needs this solved first. https://github.com/ipfs/interface-ipfs-core/issues/126